datafusion/physical-plan/src/joins/stream_join_utils.rs

```diff
@@ -155,7 +156,10 @@ impl PruningJoinHashMap {
     /// # Returns
     /// The size of the hash map in bytes.
     pub(crate) fn size(&self) -> usize {
-        self.map.allocation_info().1.size() + self.next.capacity() * size_of::<u64>()
+        let fixed_size = size_of::<PruningJoinHashMap>();
+
+        estimate_memory_size::<(u64, u64)>(self.map.capacity(), fixed_size).unwrap()
```

Comment:
maybe put a TODO here: once we updated the the latest hashbrown version, you no longer need to estimate this, see https://docs.rs/hashbrown/latest/hashbrown/struct.HashTable.html#method.allocation_size

Reply:
Yeah, good idea